VirtualReality/MRML/vtkVirtualRealityViewInteractor.h

```diff
@@ -36,8 +36,22 @@ class VTK_SLICER_VIRTUALREALITY_MODULE_MRML_EXPORT vtkVirtualRealityViewInteract
   vtkTypeMacro(vtkVirtualRealityViewInteractor,vtkOpenVRRenderWindowInteractor);
   void PrintSelf(ostream& os, vtkIndent indent);
 
+  virtual void SetInteractorStyle(vtkInteractorObserver*);
+
   virtual void RecognizeComplexGesture(vtkEventDataDevice3D* edata) override;
 
+  /// Set trigger button function
+  /// By default it is the same as grip (\sa GetButtonFunctionStringGripObjectsAndWorld)
+  /// Empty string disables button
+  void SetTriggerButtonFunction(std::string functionId);
+
+  /// Get string constant corresponding to button function "grip objects and world"
+  static std::string GetButtonFunctionStringGripObjectsAndWorld() { return "GripObjectsAndWorld"; };
```

Comment:
This name is really long. Something like GetFunctionGripObjectsAndWorld would be probably enough.

Also, I think "Grab" instead of "Grip" might be better. We should ask a native speaker.

Reply:
+1 Grab instead of Grip

Also I was going to comment that I didn't understand that function name.  Maybe clearer:
GetButtonFunctionIdForGrabObjectsAndWorld.  But really, why is this extra static function even there at all?  All it does is return the string form of part of the function name.  Plus, doing a string compare in the event dispatcher seems like wasteful.  Wouldn't an enum be more in the spirit of C++ programming?

Reply:
We thought about using enums, but we anticipate that modules will define their own operating modes and it is much easier to manage that than enums with user values. With strings, modules can prefix their string constant with the module name to avoid conflicts and the strings are also self-describing.

Reply:
Yes, the strings are self-describing, which is why I asked if the function really serves a purpose here.

Anyway it's not a big deal, just chiming in.

Reply:
The function's purpose is to make the developer easily aware of available options. Same thing as what we do in vtkSegmentationConverter with the representation names. If we have these constants listed like this, then it is easy to see what's available.
First I wanted to add an enum, but for the reasons described above, we went for strings instead.

Reply:
Just trying to avoid long compile times 😉